source/base/ByteBuffer.ooc

```diff
@@ -85,12 +80,10 @@ _SlicedByteBuffer: class extends ByteBuffer {
 	}
 }
 _RecoverableByteBuffer: class extends ByteBuffer {
-	_recover: Func (ByteBuffer)
+	_recover: Func (ByteBuffer) -> Bool
 	init: func (pointer: UInt8*, size: Int, =_recover) { super(pointer, size) }
 	free: override func {
-		if (!this _forceFree)
-			this _recover(this)
-		else {
+		if (this _recover(this)) {
```

Comment:
I understand what you are doing here, but the code like 
```ooc
if (this recover())
    delete(this)
```
is quite confusing (*if recover then delete*) :) Maybe invert the flag ?